src/framegrab/rtsp_discovery.py

```diff
@@ -95,24 +110,27 @@ def discover_onvif_devices(
             logger.debug("ONVIF device discovery disabled")
             return device_ips
 
-        wsd = WSDiscovery()
-        wsd.start()
-        types = [QName("http://www.onvif.org/ver10/network/wsdl", "NetworkVideoTransmitter")]
-        ret = wsd.searchServices(types=types)
-        for service in ret:
-            xaddr = service.getXAddrs()[0]
-            parsed_url = urllib.parse.urlparse(xaddr)
-            ip = parsed_url.hostname
-            port = parsed_url.port or 80  # Use the default port 80 if not specified
-
-            logger.debug(f"Found ONVIF service at {xaddr}")
-            device_ip = ONVIFDeviceInfo(ip=ip, port=port, username="", password="", xaddr=xaddr, rtsp_urls=[])
-
-            if auto_discover_mode is not AutodiscoverMode.ip_only:
-                RTSPDiscovery._try_logins(device=device_ip, auto_discover_mode=auto_discover_mode)
+        try:
+            wsd = RTSPDiscovery._get_wsd()
+            wsd.start()
+            types = [QName("http://www.onvif.org/ver10/network/wsdl", "NetworkVideoTransmitter")]
+            ret = wsd.searchServices(types=types)
+            for service in ret:
+                xaddr = service.getXAddrs()[0]
+                parsed_url = urllib.parse.urlparse(xaddr)
+                ip = parsed_url.hostname
+                port = parsed_url.port or 80  # Use the default port 80 if not specified
+
+                logger.debug(f"Found ONVIF service at {xaddr}")
+                device_ip = ONVIFDeviceInfo(ip=ip, port=port, username="", password="", xaddr=xaddr, rtsp_urls=[])
+
+                if auto_discover_mode is not AutodiscoverMode.ip_only:
+                    RTSPDiscovery._try_logins(device=device_ip, auto_discover_mode=auto_discover_mode)
+
+                device_ips.append(device_ip)
+        finally:
+            wsd.stop()
```

Comment:
Could you leave a comment here that while this is good style, wsd.stop seems to not work